jf_agent/config_file_reader.py

```diff
@@ -308,12 +315,50 @@ def _get_git_config_from_yaml(yaml_config) -> List[GitConfig]:
 git_providers = ['bitbucket_server', 'bitbucket_cloud', 'github', 'gitlab']
 
 
-def get_ingest_config(config: ValidatedConfig, creds, endpoint_jira_info: dict) -> IngestionConfig:
+def _get_jf_ingest_git_auth_config(
+    company_slug: str, config: GitConfig, git_creds: dict, skip_ssl_verification: bool,
+):
+    from jf_agent.git.utils import BBC_PROVIDER, BBS_PROVIDER, GH_PROVIDER, GL_PROVIDER
+
+    try:
+        if config.git_provider == BBS_PROVIDER:
+            return None
+
+        if config.git_provider == BBC_PROVIDER:
+            return None
+
+        if config.git_provider == GH_PROVIDER:
+            return JFIngestGitAuthConfig(
+                company_slug=company_slug,
+                token=git_creds['github_token'],
+                base_url=config.git_url,
+                verify=not skip_ssl_verification,
+            )
+        if config.git_provider == GL_PROVIDER:
+            return None
+
+    except Exception as e:
+        logging_helper.log_standard_error(
+            logging.ERROR, msg_args=[config.git_provider, e], error_code=2101, exc_info=True,
```

Comment:
What does this error code map to?

Reply:
This maps to "Failed to Connect", the lookup table is in the `logging_helper.py` [file in JF Ingest](https://github.com/Jellyfish-AI/jf_ingest/blob/491889ffeddc1b01aa95930bd6863e9951af64ae/jf_ingest/logging_helper.py#L101).

These Error codes were a part of JF Agent a LONG time ago, and as far as I know they aren't used meaningfully anywhere. Infact I think they do more harm than good by obscuring the code for no benefit. They're around as a legacy thing and should probably get revamped or removed